### ISSUE TITLE
Undefined index: cid

### ISSUE THREAD

**Issue by s1rc** [okta/okta-jwt-verifier-php#68]
When decoding the `id_token` there is an `aud` index in the payload but no `cid` index is present. 

Setting `setClientId(null)` results in an exception saying it is required.

Is there perhaps an Okta config missing? The odd thing is that the `aud` key in the payload matches the Okta ClientId, not `api://default`.

**Comment by bryanapellanes-okta** [comment]
@s1rc ,
Thanks for reaching out!  I will inquire internally to get you an answer to your question.

**Comment by bretterer** [comment]
Hi @s1rc, Could you please provide reproduction code for your issue?  This would help in getting you a solution.

**Comment by s1rc** [comment]
@bretterer of course.

So using this with the suggested audience:
```php
$jwtVerifier = (new \Okta\JwtVerifier\JwtVerifierBuilder())
    ->setDiscovery(new \Okta\JwtVerifier\Discovery\Oauth)
    ->setAdaptor(new \Okta\JwtVerifier\Adaptors\FirebasePhpJwt)
    ->setAudience('api://default')
    ->setClientId(config('services.okta.client_id'))
    ->setIssuer(config('services.okta.base_url') . '/oauth2/default')
    ->build();

$idToken = $jwtVerifier->verify($idTokenString);
```
Results in:
`Audience does not match what is expected. Make sure to provide the audience with `setAudience()` from the JwtVerifierBuilder.`

Since setting the audience as the clientID since that is what `aud` is representing in the `id_token` payload:
```php
$jwtVerifier = (new \Okta\JwtVerifier\JwtVerifierBuilder())
    ->setDiscovery(new \Okta\JwtVerifier\Discovery\Oauth)
    ->setAdaptor(new \Okta\JwtVerifier\Adaptors\FirebasePhpJwt)
    ->setAudience(config('services.okta.client_id'))
    ->setClientId(config('services.okta.client_id'))
    ->setIssuer(config('services.okta.base_url') . '/oauth2/default')
    ->build();

$idToken = $jwtVerifier->verify($idTokenString);
```
Results in:
`Undefined index: cid`

**Comment by abonne01** [comment]
hello,

I have the same issue. For OpenID token (not access token), cid check is not relevant for openid token. 
Then it's not relevant in this case.

```
    private function validateClaims(array $claims)
    {
        $this->validateNonce($claims);
        $this->validateAudience($claims);
        $this->validateClientId($claims);
    }
```
on function $this->validateClientId($claims); check is skiped if clientId is null.

But if we don't set it in builder, we rade this exception : 
`Fatal error: Uncaught InvalidArgumentException: Your client ID is missing. You can copy it from the Okta Developer Console in the details for the Application you created. Follow these instructions to find it: https://bit.ly/finding-okta-app-credentials in /var/www/vhosts/pocOkta/vendor/okta/jwt-verifier/src/JwtVerifierBuilder.php on line 176`

thanks for Help

**Comment by bretterer** [comment]
I know this isn't really the answer you are looking for, but, Yes, this appears to be a bug. as you found, ID Tokens do not have a cid, making the cid check cause issues.  I will be updating this soon in order to give you a `validateAccessToken` and `validateIdToken` method, with the correct checks. 

Please stay tuned to this ticket for updates.

**Comment by abonne01** [comment]
@s1rc 

when you instanciate openid ticket, you must specify 
`
->setDiscovery(new \Okta\JwtVerifier\Discovery\Oidc)
`

I saw on your example that you specified 
`
->setDiscovery(new \Okta\JwtVerifier\Discovery\Oauth)
`

**Comment by bretterer** [comment]
@s1rc and @abonne01,

I have put together a quick branch that may fix your issues.  for an ID token, just update your code to use `validateIdToken` with the same signature.

**Comment by s1rc** [comment]
@bdemers amazing so far works using:
```php
$jwtVerifier = (new \Okta\JwtVerifier\JwtVerifierBuilder())
    ->setDiscovery(new \Okta\JwtVerifier\Discovery\Oidc)
    ->setAdaptor(new \Okta\JwtVerifier\Adaptors\FirebasePhpJwt)
    ->setAudience(config('services.okta.client_id'))
    ->setClientId(config('services.okta.client_id'))
    ->setIssuer(config('services.okta.base_url') . '/oauth2/default')
    ->build();

$idToken = $jwtVerifier->verifyIdToken($idTokenString);
```

Thank you!

**Comment by abonne01** [comment]
@bdemers 
thanks it worked as well. 
waiting for a permanent correction to avoid composer erasing this on next update :)

**Comment by bretterer** [comment]
@s1rc and @abonne01 I will get a release out by end of week, possibly today if schedule allows

Thank you!

**Comment by s1rc** [comment]
Thank you @bretterer!

**Comment by abonne01** [comment]
thanks. If you could aldo add a comment in readme file for that, an example for open ID, it should be nice.

**Comment by bretterer** [comment]
Sorry for the delay here, it is now released as 1.1.0

https://github.com/okta/okta-jwt-verifier-php/releases/tag/1.1.0
